applications/ShallowWaterApplication/custom_elements/euler_prim_var_element.cpp

```diff
@@ -0,0 +1,343 @@
+        boost::numeric::ublas::bounded_matrix<double,TNumNodes*3,TNumNodes*3> aux_convect  = ZeroMatrix(TNumNodes*3,TNumNodes*3);
+        boost::numeric::ublas::bounded_matrix<double,TNumNodes*3,TNumNodes*3> aux_q_div_u  = ZeroMatrix(TNumNodes*3,TNumNodes*3);
+        boost::numeric::ublas::bounded_matrix<double,TNumNodes*3,TNumNodes*3> aux_w_grad_h = ZeroMatrix(TNumNodes*3,TNumNodes*3);
+        boost::numeric::ublas::bounded_matrix<double,TNumNodes*3,TNumNodes*3> aux_u_diffus = ZeroMatrix(TNumNodes*3,TNumNodes*3);
+        boost::numeric::ublas::bounded_matrix<double,TNumNodes*3,TNumNodes*3> aux_h_diffus = ZeroMatrix(TNumNodes*3,TNumNodes*3);
+        
+        // Loop on Gauss points. In this case, number of Gauss points and number of nodes coincides
+        for(unsigned int igauss = 0; igauss < TNumNodes; igauss++)
+        {
+            noalias(N) = row(Ncontainer, igauss);
+            
+            // Build shape and derivatives functions at Gauss points
+            for(unsigned int nnode = 0; nnode < TNumNodes; nnode++)
+            {
+                // Height gradient
+                DN_DX_height(0, 2+nnode*3) = DN_DX(nnode,0);
+                DN_DX_height(1, 2+nnode*3) = DN_DX(nnode,1);
+                // Velocity divergence
+                DN_DX_vel(0,  nnode*3) = DN_DX(nnode,0);
+                DN_DX_vel(0,1+nnode*3) = DN_DX(nnode,1);
+                // Height shape funtions
+                N_height(0,2+nnode*3) = N[nnode];
+                // Velocity shape functions
+                N_vel(0,  nnode*3) = N[nnode];
+                N_vel(1,1+nnode*3) = N[nnode];
+                // Velocity gradient
+                grad_vel_1(0,  nnode*3) = DN_DX(nnode,0);
+                grad_vel_1(1,1+nnode*3) = DN_DX(nnode,0);
+                grad_vel_2(0,  nnode*3) = DN_DX(nnode,1);
+                grad_vel_2(0,1+nnode*3) = DN_DX(nnode,1);
+            }
+            
+            noalias(mass_matrix)  += prod(trans(N_vel),N_vel);
+            noalias(mass_matrix)  += prod(trans(N_height),N_height);
+            
+            noalias(aux_convect)  += prod(trans(N_height),Matrix(prod(velocity,DN_DX_height)));
+            noalias(aux_convect)  += velocity(0,0) * prod(trans(N_vel),grad_vel_1);
+            noalias(aux_convect)  += velocity(0,1) * prod(trans(N_vel),grad_vel_2);
+            
+            noalias(aux_q_div_u)  += prod(trans(N_height),DN_DX_vel);
+            noalias(aux_w_grad_h) += prod(trans(N_vel),DN_DX_height);
+            
+            noalias(aux_u_diffus) += prod(trans(DN_DX_vel),DN_DX_vel);
+            noalias(aux_h_diffus) += prod(trans(DN_DX_height),DN_DX_height);
+        }
+        
+        // Copmute stabilization parameters
+        bool stabilization = true;
+        double Ctau = 0.005;
+        double tau_u = 0, tau_h = 0;
+        double fheight = fabs(height);
+        if (stabilization && fheight > 1e-6)
+        {
+            tau_u = Ctau/elem_length*pow(gravity/fheight,0.5);
```

Comment:
std::sqrt()... I have already seen this code, can be moved to a common utility/class?